File: programmers/모의고사/모의고사.js
function solution(answers) {
    const students = [[1,2,3,4,5], [2,1,2,3,2,4,2,5], [3,3,1,1,2,2,4,4,5,5]];
    let maxCount = -1;
    let answer = [];

    students.forEach((student, index) => {
        let count = 0;
        let flag = false;

        for(let i = 0 ; i < answers.length ; i++){
            if(answers[i] === student[i % student.length]){
                flag = true;
                count += 1;
            }
        }

        if(flag){
            if(count > maxCount){
                maxCount = count;
                answer = [index+1];
            } else if(count === maxCount){
                answer.push(index+1);
            }
        }
    });

    return answer;
}

File: programmers/모의고사/모의고사.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int[] solution(int[] answers) {
        int[] one = {1,2,3,4,5};
        int[] two = {2,1,2,3,2,4,2,5};
        int[] three = {3,3,1,1,2,2,4,4,5,5};

        long one_count = IntStream.range(0, answers.length)
            .filter(i -> answers[i] == one[i % one.length])
            .count();

        long two_count = IntStream.range(0, answers.length)
            .filter(i -> answers[i] == two[i % two.length])
            .count();

        long three_count = IntStream.range(0, answers.length)
            .filter(i -> answers[i] == three[i % three.length])
            .count();

        long max = Math.max(one_count, Math.max(two_count, three_count));
        if(max == 0) return new int[0];

        List<Integer> answer = new ArrayList<>();
        if(one_count == max) answer.add(1);
        if(two_count == max) answer.add(2);
        if(three_count == max) answer.add(3);

        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}
